### PR TITLE
Expanded ghost gas analyzer

### PR DESCRIPTION
**Pull request by Bjamcham**
## What Does This PR Do
Adds a verb `Toggle Gas Analyzer` to ghosts that when enabled allows the user to click pipes to scan their gas contents. Provides the same information as an inhand gas analyzer.

## Why It's Good For The Game
Adds a tool for ghosts that lets them better see what's going on inside of a pipe. This lets a ghost better see what's happening in a pipenet, be it debugging someone's project, checking pipes for a ticket, or just curiosity to see what a player is up to. 

## Images of changes
image

## Testing
Spawned in, used ghost examine/click as normal, toggled the analyzer, checked pipes, checked turfs, tried the analyze air verb, all worked as intended.

## Changelog
:cl:
add: Adds the verb 'Toggle Gas Analyzer' to ghosts that allows ghosts to click pipes in order to scan the pipe's gas contents.
/:cl:

### CODE REVIEW

File: code/modules/mob/dead/observer/observer_base.dm [hunk 1]
```diff
@@ -529,14 +530,27 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
 /mob/dead/observer/verb/analyze_air()
 	set name = "Analyze Air"
 	set category = "Ghost"
 
 	if(!isobserver(usr))
 		return
-
+	gas_analyze()
 	// Shamelessly copied from the Gas Analyzers
+/mob/dead/observer/proc/gas_analyze(mob/user as mob)
```
Comment (lewcc) [inline code/modules/mob/dead/observer/observer_base.dm:553]:
```suggestion
/mob/dead/observer/proc/gas_analyze(mob/user)
```

File: code/_onclick/observer_onclick.dm [hunk 2]
```diff
@@ -28,6 +28,9 @@
 		return
 	if(world.time <= next_move)
 		return
+	if(gas_analyzer && istype(A, /obj/machinery/atmospherics/pipe))
```
Comment (lewcc) [inline code/_onclick/observer_onclick.dm:31]: Honestly I'd just say to move this onto `attack_ghost()` for pipes

Comment (Bjamcham) [reply code/_onclick/observer_onclick.dm:31]: I'm assuming you mean set it up as 
```/obj/machinery/atmospherics/pipe/attack_ghost(mob/user)
	gas_analyzer()
``` 
With it setup like this though, I can't call the new proc I setup on /mob/dead/observer, and I'm not really sure where under pipes I should or would setup a new proc for anaylzing. Where I currently placed it does seem out of place though.

**Comment by Bjamcham** [comment]
Turns out this doesn't actually work and I messed up with testing and gave myself a false positive. Will reopen if I sort out the issue.

File: code/modules/mob/dead/observer/observer_base.dm [hunk 3]
```diff
@@ -32,6 +32,7 @@ GLOBAL_VAR_INIT(observer_default_invisibility, INVISIBILITY_OBSERVER)
 	var/list/data_hud_seen = list()
 	var/ghost_orbit = GHOST_ORBIT_CIRCLE
 	var/health_scan = FALSE //does the ghost have health scanner mode on? by default it should be off
+	var/gas_analyzer = FALSE //on click gas analyzer
```
Comment (DGamerL) [inline code/modules/mob/dead/observer/observer_base.dm:35]: Could you autodoc this?

**Review by Burzah — APPROVED** [review]

**Review by S34NW — APPROVED** [review]

**Pull request merged**